lib/handlers/compile.js

```diff
@@ -433,7 +433,7 @@ export class CompileHandler {
         let files;
         if (req.body.files) files = req.body.files;
 
-        if (source === undefined) {
+        if (source === undefined || Object.keys(req.body).length === 0) {
```

Comment:
Can you put this check in `checkRequestRequirements()`? that will happen earlier

Reply:
For an empty body it goes into the "api style" section of parseRequest and doesn't check the request requirements. Is there another earlier spot that would work?

Reply:
Looks like they use type-is [here](https://github.com/expressjs/express/blob/master/lib/request.js#L289), which specifies that "If the request has no body, even if there is a Content-Type header, then null is returned" in their [readme](https://github.com/jshttp/type-is#readme).

Reply:
ah I see, somehow I read something else, but obviously that was wrong. Let's keep it like this I guess